tests/extension/oneapi_kernel_compiler_spirv/kernel_compiler_spirv.cpp

```diff
@@ -0,0 +1,235 @@
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+#include <exception.hpp>
+
+namespace kernel_compiler_spirv::tests {
+
+#ifdef SYCL_EXT_ONEAPI_AUTO_LOCAL_RANGE
+
+sycl::kernel_bundle<sycl::bundle_state::executable> loadKernelsFromFile(
+    sycl::queue& q, std::string file_name) {
+  namespace syclex = sycl::ext::oneapi::experimental;
+
+  // Read the SPIR-V module from disk.
+  std::ifstream spv_stream(file_name, std::ios::binary);
+  spv_stream.seekg(0, std::ios::end);
+  size_t sz = spv_stream.tellg();
+  spv_stream.seekg(0);
+  std::vector<std::byte> spv(sz);
+  spv_stream.read(reinterpret_cast<char*>(spv.data()), sz);
+
+  // Create a kernel bundle from the binary SPIR-V.
+  sycl::kernel_bundle<sycl::bundle_state::ext_oneapi_source> kb_src =
+      syclex::create_kernel_bundle_from_source(
+          q.get_context(), syclex::source_language::spirv, spv);
+
+  // Build the SPIR-V module for our device.
+  sycl::kernel_bundle<sycl::bundle_state::executable> kb_exe =
+      syclex::build(kb_src);
+  return kb_exe;
+}
+
+void testSimpleKernel(sycl::queue& q, const sycl::kernel& kernel,
+                      int multiplier, int added) {
+  const auto num_args = kernel.get_info<sycl::info::kernel::num_args>();
+  REQUIRE(num_args == 2);
+
+  constexpr int N = 4;
+  std::array<int, N> input_array{0, 1, 2, 3};
+  std::array<int, N> output_array{};
+
+  sycl::buffer input_buffer(input_array.data(), sycl::range<1>(N));
+  sycl::buffer output_buffer(output_array.data(), sycl::range<1>(N));
+
+  q.submit([&](sycl::handler& cgh) {
+    cgh.set_arg(0, input_buffer.get_access<sycl::access::mode::read>(cgh));
+    cgh.set_arg(1, output_buffer.get_access<sycl::access::mode::write>(cgh));
+    cgh.parallel_for(sycl::range<1>{N}, kernel);
+  });
+
+  sycl::host_accessor out{output_buffer};
+  for (int i = 0; i < N; i++) {
+    CHECK(out[i] == ((i * multiplier) + added));
+  }
+}
+
+template <typename T>
+void testParam(sycl::queue& q, const sycl::kernel& kernel) {
+  const auto num_args = kernel.get_info<sycl::info::kernel::num_args>();
+  REQUIRE(num_args == 4);
+
+  // Kernel computes sum of squared inputs.
+  const T a = 2;
+  const T b = 5;
+  const T expected = (a * a) + (b * b);
+
+  sycl::buffer a_buffer(&a, sycl::range<1>(1));
+
+  T* const b_ptr = sycl::malloc_shared<T>(1, q);
+  b_ptr[0] = b;
+
+  T output{};
+  sycl::buffer output_buffer(&output, sycl::range<1>(1));
+
+  q.submit([&](sycl::handler& cgh) {
+    sycl::local_accessor<T, 1> local(1, cgh);
+    // Pass T for scalar parameter.
+    cgh.set_arg(0, a);
+    // Pass USM pointer for OpTypePointer(CrossWorkgroup) parameter.
```

Comment:
Now I realize the content of the SPIR-V binary should be described somewhere otherwise the test is not understandable. :-)